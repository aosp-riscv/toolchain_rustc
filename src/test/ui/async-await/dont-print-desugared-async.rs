// Test that we don't show variables with from async fn desugaring

// edition:2018
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
#![feature(async_await)]
=======
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)

async fn async_fn(&ref mut s: &[i32]) {}
//~^ ERROR cannot borrow data in a `&` reference as mutable [E0596]

fn main() {}
