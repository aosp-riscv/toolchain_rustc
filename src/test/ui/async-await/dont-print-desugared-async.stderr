error[E0596]: cannot borrow data in a `&` reference as mutable
<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
  --> $DIR/dont-print-desugared-async.rs:6:20
=======
  --> $DIR/dont-print-desugared-async.rs:5:20
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
   |
LL | async fn async_fn(&ref mut s: &[i32]) {}
   |                   -^^^^^^^^^
   |                   ||
   |                   |cannot borrow as mutable through `&` reference
   |                   help: consider changing this to be a mutable reference: `&mut ref mut s`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0596`.
